cores/esp32/IPAddress.cpp

```diff
@@ -364,5 +244,168 @@ bool IPAddress::fromString6(const char *address) {
     return true;
 }
 
-// declared one time - as external in IPAddress.h
-IPAddress INADDR_NONE(0, 0, 0, 0);
+IPAddress& IPAddress::operator=(const uint8_t *address)
+{
+    // IPv4 only conversion from byte pointer
+    _type = IPv4;
+    memset(_address.bytes, 0, sizeof(_address.bytes));
+    memcpy(&_address.bytes[IPADDRESS_V4_BYTES_INDEX], address, sizeof(uint32_t));
+    return *this;
+}
+
+IPAddress& IPAddress::operator=(const char *address)
+{
+    fromString(address);
+    return *this;
+}
+
+IPAddress& IPAddress::operator=(uint32_t address)
+{
+    // IPv4 conversion
+    // See note on conversion/comparison and uint32_t
+    _type = IPv4;
+    memset(_address.bytes, 0, sizeof(_address.bytes));
+    _address.dword[IPADDRESS_V4_DWORD_INDEX] = address;
+    return *this;
+}
+
+IPAddress& IPAddress::operator=(const IPAddress& address){
+    _type = address.type();
+    _zone = address.zone();
+    memcpy(_address.bytes, address._address.bytes, sizeof(_address.bytes));
+    return *this;
+}
+
+bool IPAddress::operator==(const IPAddress& addr) const {
+    return (addr._type == _type)
+        && (memcmp(addr._address.bytes, _address.bytes, sizeof(_address.bytes)) == 0);
+}
+
+bool IPAddress::operator==(const uint8_t* addr) const
+{
+    // IPv4 only comparison to byte pointer
+    // Can't support IPv6 as we know our type, but not the length of the pointer
+    return _type == IPv4 && memcmp(addr, &_address.bytes[IPADDRESS_V4_BYTES_INDEX], sizeof(uint32_t)) == 0;
+}
+
+uint8_t IPAddress::operator[](int index) const {
+    if (_type == IPv4) {
+        return _address.bytes[IPADDRESS_V4_BYTES_INDEX + index];
```

Comment:
There is no boundary check here and it is quite easy for a user to forget to check for the type before using this `operator[]`.

Not sure what you should return here if the index is out of bounds, but trying to debug a crash is also not really nice.

Reply:
Agreed and also should be fixed [here](https://github.com/arduino/ArduinoCore-API/blob/master/api/IPAddress.cpp#L290) first